File: lua/strategies/random.lua
-- A strategy that always guesses randomly.

local createRandomCode = require 'simulation'.createRandomCode

local function reset()
  return createRandomCode(CODE_LENGTH)
end

local function guess(blacks, whites)
  return createRandomCode(CODE_LENGTH)
end

return {
  reset = reset,
  guess = guess
}

File: java/src/strategy/implementations/Random.java
package strategy.implementations;

import simulation.Simulation;
import strategy.Answer;
import strategy.Code;
import strategy.IMastermindStrategy;

/**
 * A strategy that always guesses randomly.
 */
public class Random implements IMastermindStrategy {
	
	private final int CODE_LENGTH;
	
	public Random(int codeLength) {
		this.CODE_LENGTH = codeLength;
	}
	
	@Override
	public Code guess(Answer answer) {
		return Simulation.createRandomCode(this.CODE_LENGTH);
	}

	@Override
	public Code reset() {
		return Simulation.createRandomCode(this.CODE_LENGTH);
	}

	@Override
	public String toString() {
		return "Random";
	}

}
